cmd/helm/release_testing.go

```diff
@@ -64,7 +64,16 @@ func newReleaseTestCmd(cfg *action.Configuration, out io.Writer) *cobra.Command
 					client.Filters["!name"] = append(client.Filters["!name"], notName.ReplaceAllLiteralString(f, ""))
 				}
 			}
-			rel, runErr := client.Run(args[0])
+
+			rel, shutdown, runErr := client.Run(args[0])
+			defer func() {
+				if shutdownErr := shutdown(); shutdownErr != nil {
+					if returnError == nil {
+						returnError = shutdownErr
+					}
+				}
+			}()
```

Comment:
Before, the error returned was the last error that was encountered in this function `newReleaseTestCmd()`. That could have been e.g. an error while retrieving the logs.

By using `defer shutdown()` we would have changed the behavior since now a possible error of the deletion of the pods would be the error to be returned. Before, this error possibly was ignored if `rel` was nil.